pkg/vsphere/adapter.go

```diff
@@ -178,8 +177,14 @@ func (a *vAdapter) readEvents(ctx context.Context, c *event.HistoryCollector) er
 			// avoid unnecessary K8s API calls
 			skip := lastEvent == nil || lastCheckpointEventKey == lastEvent.GetEvent().Key
 			if !skip {
-				logger.Debug("creating checkpoint")
-				if err := a.KVStore.Save(ctx); err != nil {
+				var current checkpoint
+				err := a.KVStore.Get(ctx, checkpointKey, &current)
```

Comment:
this could also be in the same line as the if
`if err := a.KVStore.Get(ctx, checkpointKey, &current); err != nil ...`